tests/dummy/app/pods/components/sample-component/template.hbs

```diff
@@ -1,19 +1,52 @@
+{{! BEGIN-SNIPPET sample-component }}
 {{yield}}
 
-{{#yield-slot 'header'}}
-  <div>
-    {{yield (block-params header 'hello' 'world')}}
-  </div>
-{{/yield-slot}}
+<div class='slot blue'>
+  {{#yield-slot 'basic'}}
+    {{yield}}
+  {{/yield-slot}}
+</div>
 
-{{#yield-slot 'main'}}
-  {{yield (block-params body 'goodnight' 'moon')}}
+{{#yield-slot 'conditional-with-default'}}
+  <div class='slot pink-2'>
+    Conditional with default: {{yield}}
+  </div>
 {{else}}
-  Default {{sampleBody}}
-{{/yield-slot}}
-
-{{#yield-slot 'footer'}}
-  <div>
-    {{yield (block-params footer 'awesome' 'you')}}
+  <div class='slot pink'>
+    Conditional with default: Default
   </div>
 {{/yield-slot}}
+
+<div class='slot green'>
+  {{#yield-slot 'params' (block-params
+    (hash x='X')
+    computedValue
+  )}}
+    {{yield}}
+  {{/yield-slot}}
+</div>
+
+<div class='slot amber'>
+  {{#yield-slot 'chained-yield'}}
+    {{#internal-component}}
+      Branch: {{yield}}
+    {{/internal-component}}
+  {{/yield-slot}}
+</div>
+
+<div class='slot red actions'>
+  {{#yield-slot 'actions' (block-params
+    (action 'internal')
+  )}}
+    {{yield}}
+  {{/yield-slot}}
+</div>
+
+<div class='slot purple'>
```

Comment:
I think we should move this either before or after the pink slot so that it is more clear that the purple slot is interaction that allows changing the value of `isCondition` inside the pink slot.